src/auth.rs

```diff
@@ -72,21 +102,99 @@ impl crate::header::HeaderEntry for ApiKey {
     }
 }
 
+/// HTTP authorization scheme.
+#[derive(Eq, Hash, Copy, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+pub enum AuthorizationScheme {
+    Bearer,
+}
+
+/// NEAR JSON RPC authorization header.
+#[derive(Eq, Hash, Clone, Debug, PartialEq)]
+pub struct Authorization(AuthorizationScheme, HeaderValue);
+
+impl Authorization {
+    pub const HEADER_NAME: &'static str = "Authorization";
+
+    /// Creates a new authorization token with the bearer scheme.
+    ///
+    /// This does not perform any token-specific validation on the token.
+    ///
+    /// See the [`auth`](self) module documentation for more information.
+    pub fn bearer<T: AsRef<str>>(token: T) -> Result<Self, InvalidHeaderValue> {
+        HeaderValue::from_bytes(&[b"Bearer ", token.as_ref().as_bytes()].concat()).map(
```

Comment:
how come we're concatenating with bytes here? Also, `from_bytes` allow ASCII characters outside the standard visible characters which might not be what we want from authorization. `from_str` would only allow the visible ASCII characters
```suggestion
        HeaderValue::from_str(&["Bearer ", token.as_ref()].concat()).map(
```

Reply:
> The signature T: AsRef<str> already enforces this guarantee at the caller level. So both of these are pretty much equivalent, but we avoid constructing an owned String when concat-ing.

ahh, ignore my suggestion then

> PS: the logic of from_str doesn't use any str-specific logic. It literally just casts it back to a slice.

interesting and weird how the docs say that it accepts visible ASCII octets only?

Reply:
> interesting and weird how the docs say that it accepts visible ASCII octets only?

Yeah, it does validation at a later point, but by then it doesn't matter what the input was, it validates a byte slice eventually. https://github.com/hyperium/http/blob/34a9d6bdab027948d6dea3b36d994f9cbaf96f75/src/header/value.rs#L226

Reply:
PS: thanks for looking through